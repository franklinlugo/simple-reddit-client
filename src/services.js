import Snoowrap from 'snoowrap';

const { USER_AGENT, CLIENT_ID, CLIENT_SECRET, REFRESH_TOKEN } = process.env;

const r = new Snoowrap({
  userAgent: USER_AGENT,
  clientId: CLIENT_ID,
  clientSecret: CLIENT_SECRET,
  refreshToken: REFRESH_TOKEN,
});

export async function getTopPost() {
  const topPosts = await r.getTop({ time: 'week', limit: 5 });
  return topPosts;
}
