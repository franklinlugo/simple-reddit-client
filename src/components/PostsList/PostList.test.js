import React from 'react';
import { screen } from '@testing-library/dom';
import { renderWithProvider } from '../../utils/tests/renderWithProvider';
import PostList from './PostsList';
import { getTopPost as getTopPostMock } from '../../services';
import { postsList } from '../../utils/tests/fixtures';

jest.mock('../../services');

afterAll(() => {
  jest.clearAllMocks();
});

test('PostList should render', async () => {
  getTopPostMock.mockResolvedValueOnce(postsList);
  const { asFragment } = renderWithProvider(<PostList />);
  await screen.findByText('...loading');
  expect(asFragment()).toMatchSnapshot();
});
