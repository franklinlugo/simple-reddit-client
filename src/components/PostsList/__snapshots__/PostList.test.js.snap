// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostList should render 1`] = `
<DocumentFragment>
  .c1 {
  width: 100%;
  padding: 1rem;
  border-bottom: 1px solid;
}

.c2 {
  display: grid;
  grid-template-columns: repeat(2,min-content);
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-column-gap: 1rem;
  column-gap: 1rem;
}

.c3 {
  margin: 0;
  padding: 0;
  line-height: 1;
  white-space: nowrap;
}

.c4 {
  margin: 1rem 0;
  color: 60;
  white-space: nowrap;
}

.c5 {
  display: grid;
  grid-template-columns: 100px 1fr;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-column-gap: 1rem;
  column-gap: 1rem;
  margin-top: 1rem;
}

.c6 {
  width: 100px;
  height: 100px;
}

.c7 {
  margin: 0;
  padding: 0;
  line-height: 1;
}

.c8 {
  display: grid;
  grid-template-columns: repeat(2,min-content);
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-top: 1rem;
}

.c9 {
  display: grid;
  grid-template-columns: repeat(2,min-content);
  white-space: nowrap;
  border: 1px solid;
  cursor: pointer;
}

.c10 {
  white-space: nowrap;
}

.c0 {
  width: 500px;
  border-right: 1px solid;
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h3
          class="c3"
        >
          111 author
        </h3>
        <h5
          class="c4"
        >
          5/5/1970, 9:00:00 PM
        </h5>
      </div>
      <div
        class="c5"
      >
        <img
          class="c6"
          src="https://image1111.com"
        />
        <p
          class="c7"
        >
          111 title
        </p>
      </div>
      <div
        class="c8"
      >
        <button
          class="c9"
        >
          Dismiss Post
        </button>
        <span
          class="c10"
        >
           0 Comments
        </span>
      </div>
    </div>
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h3
          class="c3"
        >
          222 author
        </h3>
        <h5
          class="c4"
        >
          5/5/1970, 9:00:00 PM
        </h5>
      </div>
      <div
        class="c5"
      >
        <img
          class="c6"
          src="https://image222.com"
        />
        <p
          class="c7"
        >
          222 title
        </p>
      </div>
      <div
        class="c8"
      >
        <button
          class="c9"
        >
          Dismiss Post
        </button>
        <span
          class="c10"
        >
           0 Comments
        </span>
      </div>
    </div>
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h3
          class="c3"
        >
          333 author
        </h3>
        <h5
          class="c4"
        >
          5/5/1970, 9:00:00 PM
        </h5>
      </div>
      <div
        class="c5"
      >
        <img
          class="c6"
          src="https://image333.com"
        />
        <p
          class="c7"
        >
          333 title
        </p>
      </div>
      <div
        class="c8"
      >
        <button
          class="c9"
        >
          Dismiss Post
        </button>
        <span
          class="c10"
        >
           0 Comments
        </span>
      </div>
    </div>
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h3
          class="c3"
        >
          444 author
        </h3>
        <h5
          class="c4"
        >
          5/5/1970, 9:00:00 PM
        </h5>
      </div>
      <div
        class="c5"
      >
        <img
          class="c6"
          src="https://image444.com"
        />
        <p
          class="c7"
        >
          444 title
        </p>
      </div>
      <div
        class="c8"
      >
        <button
          class="c9"
        >
          Dismiss Post
        </button>
        <span
          class="c10"
        >
           0 Comments
        </span>
      </div>
    </div>
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h3
          class="c3"
        >
          555 author
        </h3>
        <h5
          class="c4"
        >
          5/5/1970, 9:00:00 PM
        </h5>
      </div>
      <div
        class="c5"
      >
        <img
          class="c6"
          src="https://image555.com"
        />
        <p
          class="c7"
        >
          555 title
        </p>
      </div>
      <div
        class="c8"
      >
        <button
          class="c9"
        >
          Dismiss Post
        </button>
        <span
          class="c10"
        >
           0 Comments
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
